### PR TITLE
Feature/rest api

### PR DESCRIPTION
**Pull request by bipinsoniguavus**
**REST API Support and pipenv support**

**Superset version**
0.27.0

**Description:** 
1. Superset is now able to connect to remote services via REST API interface using PandasConnector and taking reference from existing [PR](https://github.com/apache/incubator-superset/pull/3492)
2. Superset is now run-able with pipenv which primarily meant to provide users and developers of applications with an easy method to setup a working environment

### CODE REVIEW

File: contrib/connectors/pandas/models.py [hunk 1]
```diff
@@ -0,0 +1,940 @@
...
+                                                headers={'Authorization': auth})
+                    else:
+                        response = requests.get(url, params=self.source_parameters)
+                    response.raise_for_status()
+                    data = BytesIO(response.content)
+                else:
+                    # Local file, so just use Pandas directly
+                    data = self.source_url
+                # Read the dataframe from the response
+                self.df = self.pandas_read_method(data, **self.pandas_read_parameters)
```
Comment (arpit-agarwal) [inline contrib/connectors/pandas/models.py]: I think adding a blank line before this make it  more readable as this is not part of else. However I am not sure about lint rules here

However I do see blank line used to make code readable and communicate thsi line is in parent block

File: contrib/connectors/pandas/models.py [hunk 2]
```diff
@@ -0,0 +1,940 @@
...
+                # see https://pandas.pydata.org/pandas-docs/stable/timeseries.html#partial-string-indexing  # NOQA
+                try:
+                    if ((col_obj.is_string or col_obj.is_dttm) and
+                            not isinstance(eq, list)):
+                        eq = "'{}'".format(eq)
+                except AttributeError:
+                    # col_obj is None, probably because the col is a metric,
+                    # in which case it is numeric anyway
+                    pass
+                # query += '(\"{col}\" {op} \"{eq}\")'.format(col=col, op=op, eq=eq)
```
Comment (arpit-agarwal) [inline contrib/connectors/pandas/models.py]: any reason to keep commented code

File: contrib/connectors/pandas/views.py [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+        DeleteMixin._delete(self, pk)
+
+    @expose('/edit/<pk>', methods=['GET', 'POST'])
+    @has_access
+    def edit(self, pk):
+        """Simple hack to redirect to explore view after saving"""
+        resp = super(PandasDatasourceModelView, self).edit(pk)
+        if isinstance(resp, basestring):
+            return resp
+        return redirect('/superset/explore/pandas/{}/'.format(pk))
```
Comment (arpit-agarwal) [inline contrib/connectors/pandas/views.py]: here also we need to use url_for

File: superset/views/core.py [hunk 4]
```diff
@@ -1449,6 +1449,11 @@ def checkbox(self, model_view, id_, attr, value):
             'DruidColumnInlineView':
                 ConnectorRegistry.sources['druid'].column_class,
         }
+
+        if 'pandas' in ConnectorRegistry.sources:
+            model = ConnectorRegistry.sources['pandas'].column_class
+            modelview_to_model[model_view] = model
+
```
Comment (arpit-agarwal) [inline superset/views/core.py]: What is use of this. do we need to add ES here (current impl)?

File: contrib/connectors/pandas/models.py (hunk 1 above)
Comment (bipinsoniguavus) [reply contrib/connectors/pandas/models.py]: Not taking care these things in current PR, may be a separate US we can check the same.

**Comment by bipinsoniguavus** [comment]
To try using REST API interfacing, use the following API:
https://api.bitbucket.org/2.0/users/karllhughes 

The results can be displayed in table format

File: contrib/connectors/pandas/models.py [hunk 5]
```diff
@@ -0,0 +1,940 @@
...
+        ('week_starting_sunday', 'W-SUN'),
+        ('week_ending_saturday', 'W-SUN'),
+        ('month', 'M'),
+        ('quarter', 'Q'),
+        ('year', 'A'),
+    ])
+
+    __tablename__ = 'pandas_datasources'
+    type = 'pandas'
+    baselink = 'pandasdatasourcemodelview'  # url portion pointing to ModelView
```
[the remark this answers is not in the document]
Comment (arpit-agarwal) [reply contrib/connectors/pandas/models.py]: our choice. but whatever we decide we need to ensure it is captured. Put a note in other PR/Task for this if we are defeering

File: contrib/connectors/pandas/models.py (hunk 1 above)
Comment (arpit-agarwal) [reply contrib/connectors/pandas/models.py]: ok. I  was more interested the knowing.  We can follow where it make sense

File: superset/views/core.py (hunk 4 above)
Comment (arpit-agarwal) [reply superset/views/core.py]: add a commnet in code `TODO: `

**Pull request merged**